### ISSUE TITLE
Prevent boot message from interleaving with service output

### ISSUE THREAD

**Issue by NekkoDroid** [systemd/systemd#34448]
### Component

systemd, systemd-homed

### Is your feature request related to a problem? Please describe

Some services (most notably `sd-firstboot.service` and `sd-homed-firstboot.service`) output to the terminal during the bootup phase. This can cause log message to be interleaved with the service output/input:

```
         Starting Network Time Synchronization...
/etc/localtime written

Welcome to your new installation of Arch Linux!

Please configure your system!

-- Press any key to proceed --         Starting Record System Boot/Shutdown in UTMP...
[  OK  ] Finished Rebuild Journal Catalog.
[  OK  ] Finished Record System Boot/Shutdown in UTMP.
```

This is slightly annoying and this particular case was solved with b7c0d92. The problem remain though since user input is requested and while that is being entered a new message might be output, resulting it part of the input being on the old line, followed by the log message on the same line and then the remaining input is placed on a new line (following example not directly copied from an output but a visualization of the problem).

```
🔐 ‣ Please enter a new root password (empty to skip): ••••[  OK  ] Started Daily manpage database update.
•••••
[  OK  ] Started Daily verification of password and group files.
```

### Describe the solution you'd like

I see mostly 2 possible ways this may be addressed:

1. Somehow provide a way that services can wait for all the output to settle down, so nothing more needs to be written by the service manager until after the service completed
2. Provide some way for services to take a lock that prevents the manager from outputting to the terminal
	- Either completely suppress the messages or
	- Queue up the messages and output them after all locks have been released

### The systemd version you checked that didn't have the feature you are asking for

256

**Comment by supakeen** [comment]
This is indeed slightly annoying; here's a screenshot with systemd 258~rc3, sometimes there's enough output that the `-- Press any key to proceed --` is no longer visible:

<img width="796" height="855" alt="Image" src="https://github.com/user-attachments/assets/53182ca3-e31d-4dc7-84f3-34c53af0b087" />


I took a look at other provisioning tools and it seems at least Anaconda's `initial-setup` takes the following approach:

```
# tell systemd to stop logging to the console, to prevent it's messages    
# with interfering with the Initial Setup TUI potentially running there    
ExecStartPre=/bin/kill -SIGRTMIN+21 1
ExecStartPre=-/bin/plymouth quit
ExecStart=/usr/libexec/initial-setup/run-initial-setup
# re-enable systemd console logging once Initial Setup is done    
ExecStartPost=/bin/kill -SIGRTMIN+20 1 
```

The killing of plymouth here is also semi-important, if it stays on another 'bug' manifests in that keypresses are very sluggish/need to be pressed multiple times for them to end up at the prompt(s). This is a separate issue that I still need to look into.
